### PR TITLE
fix: widget refactor actions fetch change

### LINKED ISSUE #7627: [Bug] Refactor of widget name doesn't update in APIs
## Description
Widget name refactor doesn't update values in API pane

### Steps to reproduce the behaviour:
1. Create any widget Text1
2. create API and give key as [{{Text1.text}}](url)
3. Change Text1 to text_box
4. Observe API pane value remains as  [{{Text1.text}}]

https://user-images.githubusercontent.com/7846888/133982923-7f7766dd-39d6-406e-ba10-7a49b0582e0b.mov

### Important Details

- Version: [Cloud / Self-Hosted vx.x]
- OS: [e.g.MacOSX]
- Environment [production, release, deploy preview]

### PR DESCRIPTION
**Pull request by ApekshaBhosale**
## Description

Widget refactor was not reflected in API pane

Fixes #7627

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fetch-actions-after-update-canvas 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.8 **(0)** | 36.55 **(-0.01)** | 33.93 **(0)** | 55.35 **(0)**
 :green_circle: | app/client/src/sagas/ActionSagas.ts | 10.88 **(0.03)** | 0.78 **(0)** | 3.7 **(0)** | 13.14 **(0.04)**
 :green_circle: | app/client/src/sagas/JSActionSagas.ts | 16.17 **(0.1)** | 2.04 **(0)** | 6.67 **(0)** | 19.12 **(0.14)**
 :green_circle: | app/client/src/sagas/PageSagas.tsx | 23.95 **(0.08)** | 5.95 **(0)** | 14.29 **(0)** | 25.78 **(0.08)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.47 **(-0.24)** | 38.26 **(-0.87)** | 36.21 **(0)** | 55.08 **(-0.27)**</details>

### CODE REVIEW

File: app/client/src/sagas/PageSagas.tsx [hunk 1]
```diff
@@ -815,6 +815,8 @@ export function* updateCanvasWithDSL(
     widgets: normalizedWidgets.entities.canvasWidgets,
   };
   yield put(initCanvasLayout(canvasWidgetsPayload));
+  yield put(fetchActionsForPage(pageId));
```
Comment (hetunandu) [inline app/client/src/sagas/PageSagas.tsx:818]: We will call this twice when loading an app for the first time? This is possibly quite wasteful

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/sagas/PageSagas.tsx (hunk 1 above)
Comment (ApekshaBhosale) [reply app/client/src/sagas/PageSagas.tsx:818]: this was here before it just got removed.   @hetunandu

**Review by hetunandu — DISMISSED** [review]

**Review by arunvjn — APPROVED** [review]

**Comment by arunvjn** [comment]
Actions are retrieved twice after refactor @ApekshaBhosale

**Comment by ApekshaBhosale** [comment]
> Actions are retrieved twice post widget name refactor @ApekshaBhosale

those 2 are actions and collections/actions i.e js collections

**Pull request merged**